### ISSUE TITLE
Implement parent state

### ISSUE THREAD

**Issue by gzip** [gmarty/scumm-nes#10]
I think this is the problem with the cassette tape mentioned here:
https://github.com/gmarty/scumm-nes/issues/4#issuecomment-2105746076

The parent object index is stored at `0x0a` in the object data. This is a one (not zero) based index to an object in the same room. It seems to be used primarily to hide objects that are behind doors like the refrigerator or the cabinet. The parent state is also stored in the upper 3 bits of byte `0x08` in the object data, but it seems to always have a value of 4 when present (there two instances of 7 in the code but not in combination with the parent index byte) , so not sure it really matters.

Basically the object needs to be hidden when the parent object is in the closed state, at least for the initial rendering.

Looks like this is also affecting the arcade room.

**Comment by gmarty** [comment]
This is one of the values that are parsed, but unused in the UI.
It would be good to signal a parent object by identing its children in the UI:
* Parent object
  * Child object 1
  * Child object 2

Then the user could drag and drop a child object to another parent in edit mode.

**Comment by gmarty** [comment]
I implemented the object identation in 1111b81 to show the relationship between parent and child. This makes it clearer to understand what's going on.
I still need to work on disabling the child states depending on parent's.
